File: 1. Two Sum/Solution.scala
import util.control.Breaks._
object Solution {
    def twoSum(nums: Array[Int], target: Int): Array[Int] = {
        var left = 0
        var right = 0
        var map = new scala.collection.mutable.HashMap[Int, Int]()
        breakable(
            for(i <- nums.indices){
                val x = target - nums(i)
                if(map.contains(x)){
                    left = map(x)
                    right = i
                    break()
                }
                map += (nums(i) -> i)
            }
        )
        Array(left, right)
    }
}

File: 1. Two Sum/Solution.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] res = new int[2];

        HashMap<Integer, Integer> hashMap = new HashMap<Integer, Integer>();
        for(int i = 0; i < nums.length; i++){
            if(hashMap.containsKey(target - nums[i])){
                res[0] = hashMap.get(target - nums[i]);
                res[1] = i;
                break;
            }else{
                hashMap.put(nums[i], i);
            }
        }

        return res;
    }
}
